src/app/campaign/nav/campaign-nav.component.ts

```diff
@@ -19,20 +20,56 @@ import { Campaign, FlightState } from '../store/models';
       >
         <span matLine>{{ flight.localFlight?.name }}</span>
         <mat-icon color="warn" *ngIf="!allocationStatusOk(flight)">priority_high</mat-icon>
-      </a>
+        <button class="menuBtn" mat-icon-button disableRipple [matMenuTriggerFor]="flightMenu" [disabled]="!isActiveFlight(flight)">
+          <mat-icon aria-label="Campaign Menu">more_vert</mat-icon>
+        </button>
+      </mat-list-item>
     </mat-nav-list>
     <a class="secondary-link" routerLink="" (click)="createFlight.emit()"><mat-icon>add</mat-icon> Add a Flight</a>
+    <mat-menu panelClass="menuPanel" #flightMenu="matMenu">
+      <button mat-menu-item (click)="onDuplicate()">
+        <mat-icon aria-hidden>file_copy</mat-icon>
+        <span>Duplicate Flight</span>
+      </button>
+      <button class="menuItem--warn" mat-menu-item (click)="onDelete()" *ngIf="canDeleteActiveFlight">
+        <mat-icon aria-hidden>delete</mat-icon>
+        <span>Delete Flight</span>
```

Comment:
I think this should toggle to "Restore Flight" if you've soft deleted it (clicked this, but haven't saved yet).

In fact, clicking this twice _does_ restore the flight.

Reply:
Great idea! The fact this is a toggle wen right over my head. Change has been pushed.